tests/har.spec.ts

```diff
@@ -281,3 +281,65 @@ it('should have popup requests', async ({ contextFactory, server }, testInfo) =>
   expect(entries[1].request.url).toBe(server.PREFIX + '/one-style.css');
   expect(entries[1].response.status).toBe(200);
 });
+
+it('should have connection details', async ({ contextFactory, server, browserName, platform }, testInfo) => {
+  const { page, getLog } = await pageWithHar(contextFactory, testInfo);
+  await page.goto(server.EMPTY_PAGE);
+  const log = await getLog();
+  const { serverIPAddress, _serverPort: port, _securityDetails: securityDetails } = log.entries[0];
+  expect(serverIPAddress).toMatch(/^127\.0\.0\.1|\[::1\]/);
+  expect(port).toBe(server.PORT);
+  expect(securityDetails).toEqual({});
```

Comment:
should we maybe make it that it returns null if there are no security details / its no secure connection?

Reply:
That's what `filterEmpties` was doing before (well, `undefined` instead of `null`). I'll bring that behavior back without adding back `filterEmpties`.

Reply:
Hmm, sorry my bad. I was confused since I thought its a method which users will consume like PPTR [has](https://pptr.dev/#?product=Puppeteer&version=v10.0.0&show=api-httpresponsesecuritydetails). Feel free to do any now what is compliant with the spec.

Reply:
This section does not have a spec (or much precedent) to my knowledge, so it comes down to what we prefer. I've put it back to the empty object `{}` since that simplifies our code and saves you from doing an extra null/undefined check when consuming the HAR. (This is a non-issue in JS/TS with the details?.foo notation, but for Python it's nice to assume the object is there and just check for a property. ie. just `details.get("foo")` and not `(deatils || {}).get("foo")`)

This does, however mean, there will be a bunch of empty objects in the har for non-secure connections.

/cc @JoelEinbinder